x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -1384,4 +1392,34 @@ protected LogicalPlan rule(Aggregate aggregate) {
             return plan;
         }
     }
+
+    /**
+     * Verify that a {@link LogicalPlan} can be executed.
+     *
+     * @param plan The logical plan to be verified.
+     * @return a collection of verification failures; empty if and only if the plan is valid.
```

Comment:
This shouldn't be @return; I'd just mention this throws a VerificationException.